src/ride_build.cpp

```diff
@@ -182,19 +182,33 @@ bool RideBuildWindow::CanPlaceShop(const ShopType *selected_shop, const XYZPoint
 
 	/* 1. Can the position itself be used to build a shop? */
 	if (_world.GetTileOwner(pos.x, pos.y) != OWN_PARK) return false;
-	auto check_if_location_is_suited = [&pos, selected_shop](const int8 offset, const bool need_flat_ground) {
+	auto check_if_location_is_suited = [&pos, selected_shop](const bool is_flat_ground) {
```

Comment:
The lambda function explodes somewhat in size.

Maybe make a new normal function for it? I don't know what Doxygen does with this, but I am guessing it won't get listed as a function. This is a shame, as `check_if_location_is_suited` is a quite nice abstraction, that may be useful at other places.